src/executables/csql.c

```diff
@@ -1460,6 +1464,205 @@ csql_do_session_cmd (char *line_read, CSQL_ARGUMENT * csql_arg)
+	    {
+	      error_code = clogin_user (user_name);
+	    }
+	  else
+	    {
+	      return DO_CMD_FAILURE;
+	    }
+	  if (error_code == NO_ERROR)
+	    {
+	      csql_new_arg.user_name = strdup (user_name);
+	      csql_new_arg.db_name = strdup (csql_arg->db_name);
+	      if (p == NULL || p[0] == '\0')
+		{
+		  csql_new_arg.passwd = (char *) NULL;	/* to fit into db_login protocol */
+		}
+	      else
+		{
+		  csql_new_arg.passwd = strdup (p);
+		}
+
+	      memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+	      fprintf (csql_Output_fp, "CONNECTED : %s %s\n", csql_arg->user_name, csql_arg->db_name);
+	    }
+
+	}
+      else
+	{
+#if defined(CS_MODE)
+	  memset (boot_Host_connected, 0, sizeof (boot_Host_connected));
+#endif /* CS_MODE */
+
+	  /*Failed to access other host or db and then access formal db_name */
+	  if (db_name == NULL)
+	    {
+	      db_name = strdup (csql_arg->db_name);
+	    }
+
+	  if (csql_Database_connected)
+	    {
+	      csql_Database_connected = false;
+	      db_end_session ();
+	      db_shutdown ();
+	    }
+	  er_init ("./csql.err", ER_NEVER_EXIT);
+	  if (db_restart_ex (UTIL_CSQL_NAME, db_name, user_name, NULL, NULL, db_get_client_type ()) != NO_ERROR)
+	    {
+	      p = getpass ((char *) csql_get_message (CSQL_PASSWD_PROMPT_TEXT));
+
+	      /* try again */
+	      if (db_restart_ex (UTIL_CSQL_NAME, db_name, user_name, p, NULL, db_get_client_type ()) != NO_ERROR)
+		{
+		  csql_Error_code = CSQL_ERR_SQL_ERROR;
+		  csql_check_server_down ();
+
+		  fprintf (csql_Output_fp, "SQL session is disconnected\n");
+		  return DO_CMD_FAILURE;
+		}
+	      else
+		{
+		  /*login success with password */
+
+		  csql_new_arg.user_name = strdup (user_name);
+		  csql_new_arg.db_name = strdup (db_name);
+		  memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+
+		  if (p[0] == '\0')
+		    {
+		      csql_arg->passwd = (char *) NULL;	/* to fit into db_login protocol */
+		    }
+		  else
+		    {
+		      csql_arg->passwd = strdup (p);
+		    }
+
+		  if (csql_arg->sysadm && au_is_dba_group_member (Au_user))
+		    {
+		      au_sysadm_disable ();
+		    }
+		  csql_Database_connected = true;
+
+		  if (csql_arg->trigger_action_flag == false)
+		    {
+		      db_disable_trigger ();
+		    }
+
+		  fprintf (csql_Output_fp, "CONNECTED : %s %s\n", csql_arg->user_name, csql_arg->db_name);
+		}
+	    }
+	  else
+	    {
+	      /*login success without password */
+	      csql_new_arg.user_name = strdup (user_name);
+	      csql_new_arg.db_name = strdup (db_name);
+	      csql_arg->passwd = (char *) NULL;
+	      memcpy (csql_arg, &csql_new_arg, sizeof (CSQL_ARGUMENT));
+
+	      if (csql_arg->sysadm && au_is_dba_group_member (Au_user))
+		{
+		  au_sysadm_disable ();
+		}
+	      csql_Database_connected = true;
+
+	      if (csql_arg->trigger_action_flag == false)
+		{
+		  db_disable_trigger ();
+		}
+
+	      fprintf (csql_Output_fp, "CONNECTED : %s %s\n", csql_arg->user_name, csql_arg->db_name);
```

Comment:
fprintf (csql_Output_fp, "Connected.\n");
is sufficient

Reply:
Ok. thank you.